### ISSUE TITLE
Move SSO Permission sets to MP from Root

### ISSUE THREAD

**Issue by davidkelliott** [ministryofjustice/modernisation-platform#5659]
### User Story

As an MP engineer
I need to run single sign on bootstrap
Without it failing due to API limits

### Value / Purpose

By moving the SSO permissions sets we use from Root account to MP, we won't need to do a data source which is going over the SSO API limits, as we can query the state of the MP TF which holds the sso permission sets.

### Proposal

Move these permission sets - https://github.com/ministryofjustice/modernisation-platform/blob/main/terraform/environments/bootstrap/single-sign-on/main.tf#L14

To MP

### Unknowns 

Need to decide where to put them

### Definition of Done

Example - [ ] Documentation has been written / updated
- [x] Permission sets moved
- [x] bootstrap single sign on job can now run in parallel 
- [x] Another team member has reviewed
- [x] Tests are green

**Comment by ep-93** [comment]
I'd be interested in helping with this for sure.

**Comment by richgreen-moj** [comment]
PR in draft https://github.com/ministryofjustice/modernisation-platform/pull/6047

**Comment by richgreen-moj** [comment]
New PR raised https://github.com/ministryofjustice/modernisation-platform/pull/6081 to reference the MP permission set resources that have now been migrated in to MP state.

**Comment by richgreen-moj** [comment]
https://github.com/ministryofjustice/modernisation-platform/pull/6086 raised to further reduce the amount of data calls made to the sso api.

**Comment by richgreen-moj** [comment]
https://github.com/ministryofjustice/modernisation-platform/pull/6097 added. new workflow to run the new single-sign-on TF folder in the MP repo

**Comment by richgreen-moj** [comment]
@davidkelliott removed the relevant sso permission set resources form the root account state and then merged https://github.com/ministryofjustice/aws-root-account/pull/859 which removes these resources from the Terraform as they are now managed in the MP repo.

**Comment by richgreen-moj** [comment]
Summary of changes made:

- Created new 'single-sign-on' TF directory to manage SSO permissions sets in the MP repo
- Migrated MP-specific SSO permission set resources from root account state to MP state
- Updated data calls for permissions sets in the bootstrap code to query outputs from MP remote state rather than AWS SSO API (which is getting throttled) - we are still doing data calls to the SSO API for the generic permissions sets but these are ran once and stored in to outputs for the bootstrap code to query
- Created new workflow to manage the new SSO TF directory in MP Repo
- Removed MP-specific SSO permission set resources from root account terraform/state
